test/jdk/javax/swing/border/Test4129681.java

```diff
@@ -21,52 +21,65 @@
  * questions.
  */
 
-import java.awt.event.ItemEvent;
-import javax.swing.BorderFactory;
-import javax.swing.Box;
-import javax.swing.JCheckBox;
-import javax.swing.JComponent;
-import javax.swing.JLabel;
-
 /*
  * @test
  * @bug 4129681
  * @summary Tests disabling of titled border's caption
- * @library /java/awt/regtesthelpers
- * @build PassFailJFrame
- * @run main/manual Test4129681
+ * @run main Test4129681
  */
 
+import java.awt.Color;
+import java.awt.Dimension;
+import java.awt.Graphics2D;
+import java.awt.Point;
+import javax.swing.BorderFactory;
+import javax.swing.JLabel;
+import javax.swing.UIManager;
+
+import javax.imageio.ImageIO;
+import java.awt.image.BufferedImage;
+import java.io.File;
+
 public class Test4129681 {
+    private static JLabel label;
+    private static final Color labelDisableColor = Color.RED;
+    private static final Dimension SIZE = new Dimension(100, 40);
+    private static final Point startPoint = new Point(8, 4);
+    private static final Point endPoint = new Point(18, 14);
+
     public static void main(String[] args) throws Exception {
-        String testInstructions = """
-                Click the checkbox to disable the label.
-                The test passes if the title of the border
-                is disabled as well as the label.
-                """;
+        int correctColoredPixels = 0;
+        int totalPixels = 0;
+        int tolerance = 20;
 
-        PassFailJFrame.builder()
-                .title("Test Instructions")
-                .instructions(testInstructions)
-                .rows(4)
-                .columns(25)
-                .splitUI(Test4129681::init)
-                .build()
-                .awaitAndCheck();
-    }
+        UIManager.setLookAndFeel("javax.swing.plaf.metal.MetalLookAndFeel");
```

Comment:
Is the test exclusive for Metal Look And Feel or should it be checked in other LAFs as well?
I think it may not work for Nimbus of GTK LAF as setting the property `UIManager.getDefaults().put("Label.disabledForeground", labelDisableColor);` may not work.

Reply:
I meant the label position might vary slightly for different L&F, here I have hardcoded it for metal L&F...... Not sure it would remain same for all the L&F......

Reply:
Since you are painting to buffered image, I don't think it should matter.